tools/package

```diff
@@ -0,0 +1,236 @@
+        # We already shifted for the name, but not for the value
+        shift_value="true"
+    else
+        # We only need one shift for both key and value
+        shift_key="true"
+    fi
+    case $lhs in
+        -out-dir | --out-dir )
+            out_dir="$rhs"
+            has_value="true"
+            ;;
+        -pg-version | --pg-version )
+            pg_version="$rhs"
+            has_value="true"
+            ;;
+        -pg-config | --pg-config )
+            pg_config="$rhs"
+            has_value="true"
+            if [ ! -f "${pg_config}" ]; then
+                echo "Invalid --pg-config: no such file"
+                exit 2
+            fi
+            ;;
+        -type | --type )
+            pkg_type="$rhs"
+            has_value="true"
+            ;;
+        -version | --version )
+            version="$rhs"
+            has_value="true"
+            ;;
+        -help | --help | -h)
+            usage
+            exit 2
+            ;;
+        *)
+            echo "unknown option: $1"
+            usage
+            exit 2
+            ;;
+    esac
+
+    if [ "$shift_key" = "true" ]; then
+        shift
+    fi
+    if [ "$has_value" = "true" ] && [ "$shift_value" = "true" ]; then
+        shift
+    fi
+done
+
+if ! command -v cargo; then
+    echo "Must have cargo installed! Check your PATH"
+    exit 2
+fi
+
+if ! command -v jq; then
+    echo "Must have jq installed! Check your PATH"
+    exit 2
+fi
+
+if ! command -v fpm; then
+    echo "Must have fpm installed! Check your PATH"
+    exit 2
+fi
+
+
+if [ -z "${version}" ]; then
+    if ! version="$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[0].version')"; then
+        echo "${version}"
+        echo "Failed to read extension version from Cargo metadata!"
+        exit 1
+    fi
+fi
+
+if [ -z "${pg_version}" ] && [ -z "${pg_config}" ]; then
+    echo "Must provide either --pg-version or --pg-config!"
+    exit 2
+fi
+
+mkdir -p "${out_dir}"
+if [ ! -d "${out_dir}" ]; then
+    echo "Invalid --out-dir '${out_dir}', directory doesn't exist!"
+    exit 2
+fi
+
+if [ -n "${pg_config}" ]; then
+    if config_version="$("${pg_config}" --version | cut -d' ' -f2 | cut -d '.' -f1)"; then
+        if [ -n "$pg_version" ]; then
+            if [[ ! "$config_version" = "$pg_version" ]]; then
+                echo "Mismatched --pg-config and --pg-version"
+                echo "  Expected version:   ${pg_version}"
+                echo "  Got from pg_config: ${config_version} "
+                exit 1
+            fi
+        else
+            pg_version="${config_version}"
+        fi
+    else
+        echo "$config_version"
+        echo "Unable to obtain version from pg_config!"
+        exit 1
+    fi
+fi
+
+# If no package format was set, select an appropriate one for the current OS
+if [ -z "$pkg_type" ]; then
+    case "$os_name" in
+        debian|ubuntu)
+            pkg_type="deb"
+            ;;
+        rhel|centos|fedora)
```

Comment:
Does this actually work with RHEL or Fedora?

Reply:
Yes, I've tested on both.

Reply:
Hmmm, okay. Maybe I misunderstood how this is supposed to work. I assumed that it would be possible to build this on fedora 35, like so:

```
> docker buildx build --platform linux/amd64 --load \                    ~/Development/promscale_extension
        --build-arg OS_NAME=fedora \
        --build-arg OS_VERSION=35 \
        --build-arg PG_VERSION=14 \
        --build-arg RUST_VERSION=1.57.0 \
        -t promscale-extension-package:0.3.1-fedora35-pg14 \
        -f dist/rpm.dockerfile \
        .
        
```

Reply:
Ah oops, forgot to update the rpm dockerfile with my fedora tweaks. Fixed.

Easiest way to build a target is `OS_NAME=fedora OS_VERSION=35 PG_VERSION=14 ARCH=x86_64 make release`, little easier to type out than the full `docker buildx build` invocation :)